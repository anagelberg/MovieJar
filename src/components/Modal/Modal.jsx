import "./Modal.scss";

function Modal({ headerContent, bodyContent, footerContent, show }) {
    if (!show) return null;

    return (
        <div className="modal" role="dialog">
            <div className="modal__content">
                <div className="modal__top">
                    <div className="modal__header">
                        {headerContent()}
                    </div>

                    <div className="modal__body">
                        {bodyContent()}
                    </div>
                </div>

                {footerContent && (
                    <div className="modal__footer">
                        {footerContent()}
                    </div>
                )}
            </div>
        </div>
    );
}

export default Modal;