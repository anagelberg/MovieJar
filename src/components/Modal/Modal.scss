@use "../../styles/_mixins.scss" as *;
@use "../../styles/_variables.scss" as *;

.modal {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.7); //change to instock black
  z-index: 10;
  @include flex-center;

  &__content {
    display: flex;
    @include side-padding;
    @include vertical-padding;
    flex-direction: column;
    color: $dark-text;
    justify-content: space-between;
    background: $light-background;
    border-radius: $border-radius;
    height: 90vh;
    width: 90vw;

    @include tablet {
      max-width: 500px;
      padding-left: 64px;
      padding-right: 64px;
      width: auto;
      height: auto;
      max-height: 262px;
    }
  }

  &__body {
    display: flex;
    flex-direction: column;
    height: 100%;
    gap: 1rem;
    @include bottom-margin;
  }

  &__top {
    height: 100%;
  }

  &__footer {
    @include flex-center($justify-content: space-between);
    @include flex-gap;
  }

  &__header {
    @include flex-center($justify-content: flex-end);
  }
}
