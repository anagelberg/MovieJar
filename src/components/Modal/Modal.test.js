import { render, screen } from '@testing-library/react';
import Modal from './Modal';

describe('Modal Component', () => {
    const headerContent = () => <div>Header Content</div>;
    const bodyContent = () => <div>Body Content</div>;
    const footerContent = () => <div>Footer Content</div>;

    it('does not render when show is false', () => {
        render(<Modal show={false} headerContent={headerContent} bodyContent={bodyContent} footerContent={footerContent} />);
        expect(screen.queryByText('Header Content')).toBeNull();
        expect(screen.queryByText('Body Content')).toBeNull();
        expect(screen.queryByText('Footer Content')).toBeNull();
    });

    it('renders correctly when show is true', () => {
        render(<Modal show={true} headerContent={headerContent} bodyContent={bodyContent} footerContent={footerContent} />);
        expect(screen.queryByText('Header Content')).toBeInTheDocument();
        expect(screen.queryByText('Body Content')).toBeInTheDocument();
        expect(screen.queryByText('Footer Content')).toBeInTheDocument();
    });

    it('renders correctly without footerContent', () => {
        render(<Modal show={true} headerContent={headerContent} bodyContent={bodyContent} />);
        expect(screen.queryByText('Header Content')).toBeInTheDocument();
        expect(screen.queryByText('Body Content')).toBeInTheDocument();
        expect(screen.queryByText('Footer Content')).toBeNull();
    });

    it('matches snapshot', () => {
        const { container } = render(<Modal show={true} headerContent={headerContent} bodyContent={bodyContent} footerContent={footerContent} />);
        expect(container).toMatchSnapshot();
    });
});
