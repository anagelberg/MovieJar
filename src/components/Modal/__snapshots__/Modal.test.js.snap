// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshot 1`] = `
<div>
  <div
    class="modal"
    role="dialog"
  >
    <div
      class="modal__content"
    >
      <div
        class="modal__top"
      >
        <div
          class="modal__header"
        >
          <div>
            Header Content
          </div>
        </div>
        <div
          class="modal__body"
        >
          <div>
            Body Content
          </div>
        </div>
      </div>
      <div
        class="modal__footer"
      >
        <div>
          Footer Content
        </div>
      </div>
    </div>
  </div>
</div>
`;
